components/rag-wrapper/src/rag_wrapper/rag_ops/qdrant_rag_ops.py

```
@@ -52,148 +38,57 @@ def __init__(
-                f"Failed to initialize Qdrant RAG operations: {e}"
-            )
-            raise
+        """Initialize Qdrant vector store and load the RAG index only if it already exists.
+
+        If the collection does not exist this will create it (using the embedding
+        LLM to determine vector dimension) and initialize the vector store but not
+        create an in-memory RAG index. If the collection exists it will connect
+        and load the VectorStoreIndex.
+        """
+        if not await self.index_exists():
+            self.logger.info(f"Collection {self.collection_name} does not exist. Use create_index() to create a new collection.")
```

Ruff (0.15.17)

[warning] Logging statement uses f-string

(G004)